core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -465,6 +478,74 @@ public static GeoPoint parseGeoPoint(String data, GeoPoint point) {
         }
     }
 
+    public static GeoBBox parseBoundingBox(XContentParser parser) throws IOException, ElasticsearchParseException {
+        XContentParser.Token token = parser.currentToken();
+        if (token != XContentParser.Token.START_OBJECT) {
+            throw new ElasticsearchParseException("failed to parse bounding box. Expected start object but found [{}]", token);
+        }
+
+        double top = Double.NaN;
+        double bottom = Double.NaN;
+        double left = Double.NaN;
+        double right = Double.NaN;
+
+        String currentFieldName;
+        String fieldName = null;
+        GeoPoint sparse = new GeoPoint();
+        EnvelopeBuilder envelope = null;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+                token = parser.nextToken();
+                if (FIELD_FIELD.match(currentFieldName)) {
+                    fieldName = parser.text();
```

Comment:
Should this inner `field` parameter really still supported? I cannot find it mentioned in the docs and it doesn't seem to get written when using one of our builders. As far as I understand the whole bbox json should look like this:
```
"some.field.foo" : {
                        ... various formats defining the box ...
}
```

This is probably legacy and should be removed (probably in another PR?)